### PR TITLE
[QUICKFIX] - Error link fixed for non-connected repos

### PR DESCRIPTION
**Pull request by krazziekay**
**What's in this PR?**
- Condition added for `findForRepoNameAndOwner` for case where the repo names doesn't exist in our DB.
  - This is for the case when the repo is not synced, meaning does not exist in RepoSyncState, but the repo owner would probably exist. So simply returning the subscription for the repo owner.

**Why**
- To avoid the undefined error when repo names doesn't exist

**How has this been tested?**  
- Test cases

**Whats Next?**
- Merging

### CODE REVIEW

**Review by gxueatlassian — COMMENTED** [review]
It seems getting strange from code flowing point of view. 
1. We have all the details (`sub`, `gitHubApp`), pull down the `servers` in `create-branch-options`, render the hbs, and allow user to pick.
2. And when user submit the form, we go all the way back from the string base `owner/repoName` to find that `gitHubInstallationId` from `Subscription` again? 

Don't we already know which `Subscription` and `GitHubServerApp` when user pick a repo and submit a new branch?

**Comment by krazziekay** [comment]
> It seems getting strange from code flowing point of view.
> 
> 1. We have all the details (`sub`, `gitHubApp`), pull down the `servers` in `create-branch-options`, render the hbs, and allow user to pick.
> 2. And when user submit the form, we go all the way back from the string base `owner/repoName` to find that `gitHubInstallationId` from `Subscription` again?
> 
> Don't we already know which `Subscription` and `GitHubServerApp` when user pick a repo and submit a new branch?


We have `gitHubAppConfig`, but don't have **subscription**. We only know the **subscription** once the user has selected a **repository** in the create branch form. And even in that case, we only know the **subscription** if the **repository** is connected/synced to our app. 
So, if the user types in a non-synced **repository**, then we don't have any data for that in `RepoSyncState` nor `Subscription`, so it was returning `undefined.` This change in query is to fix that, so that we're sending the **subscription** based on just repo owner for non-connected/non-synced repos.

File: src/models/subscription.ts [hunk 1]
```diff
@@ -154,15 +155,30 @@ export class Subscription extends Model {
...
-			"SELECT * " +
+		let query = "SELECT * " +
 			"FROM \"Subscriptions\" s " +
 			"LEFT JOIN \"RepoSyncStates\" rss on s.\"id\" = rss.\"subscriptionId\" " +
 			"WHERE s.\"jiraHost\" = :jiraHost " +
-			"AND rss.\"repoName\" = :repoName " +
-			"AND rss.\"repoOwner\" = :repoOwner",
+			"AND rss.\"repoOwner\" = :repoOwner ";
+
+		const repo = await RepoSyncState.findAll({
```
Comment (gxueatlassian) [inline src/models/subscription.ts:164]: `RepoSyncState.findAll` will load all repos into js memory. Do we need that?

**Review by gxueatlassian — CHANGES_REQUESTED** [review]
Feels like an old PR? Do we still need this? Happy to reopen if needed.

**Pull request closed without merging**